documentation/reference/document-v1-api-reference.html

```diff
@@ -1,212 +1,342 @@
+    See <a href="../routing.html">routes</a>.</td>
+</tr><tr id="selection">
+  <th>selection</th>
+  <td>String</td>
+  <td>Select a set of documents using <a href="../content/visiting.html">visiting</a> -
+    details in <a href="document-select-language.html">document selector language</a>.</td>
+</tr><tr id="timeout">
+  <th>timeout</th>
+  <td>String</td>
+  <td>Request timeout in seconds, or with optional ks, s, ms or µs unit. Default value is 175s.
+    Visitor timeouts will be set to 5s less than this.</td>
+</tr><tr id="tracelevel">
+  <th>tracelevel</th>
+  <td>Integer</td>
+  <td>Number in the range [0,9], where higher gives more details.
+    The trace dumps which nodes and chains the document operation has touched.
+    See <a href="../routing.html">routes</a>.</td>
+</tr><tr id="wanteddocumentcount">
+  <th>wantedDocumentCount</th>
+  <td>Integer</td>
+  <td>
+    <p>
+      Best effort attempt to not respond to the client before <code>wantedDocumentCount</code>
+      number of documents can be returned.
+      Response may still contain fewer documents if there are not enough matching documents left
+      to visit in the cluster, or if the visiting times out.
+      This parameter is intended for the case when you have relatively few documents in your cluster
+      and where each GET operation otherwise would only return a handful of documents.
+    </p><p>
+      Note that the maximum value of <code>wantedDocumentCount</code> is bounded
+      by an implementation-specific limit to prevent excessive resource usage.
+      If the cluster has many documents (on the order of tens of millions),
+      there is no need to set this value.
+    </p>
+  </td>
+</tr>
 </tbody>
 </table>
+
+
+
+<h2 id="request-body">Request body</h2>
+<p>
+  POST and PUT has a body.
+  A field has a <em>value</em> for a POST and an <em>update operation object</em> for PUT.
+  Refer to the <a href="document-json-format.html">document format</a>:
+<pre>
+{
+    "fields": {
+        "&lt;fieldname&gt;": "&lt;value&gt;",
+        ...
+    }
+}
+</pre><pre>
+{
+    "fields": {
+        "&lt;fieldname&gt;": {
+            "&lt;update-operation&gt;" : "&lt;value&gt;",
+        }
+        ...
+    }
+}
+</pre>
+</p>
+
+
+
+<h2 id="http-status-codes">HTTP status codes</h2>
+Non-exhaustive list of status codes:
+<table class="table">
+<tr>
+  <th>200</th>
+  <td>OK. Note that updating a non-existing document is defined as success.</td>
+</tr><tr>
+  <th>404</th>
+  <td>Not found.</td>
+</tr><tr>
+  <th>412</th>
+  <td><a href="#condition">condition</a> is not met.
+    Also inspect the <code>message</code> field for details.</td>
+</tr><tr>
+  <th>504</th>
+  <td>Gateway timeout.</td>
+</tr>
+</table>
+
+
+
+<h2 id="response-format">Response format</h2>
+<p>
+  Documents and operations use the <a href="document-json-format.html">document format</a>.
+  The document fields must match the <a href="../schemas.html">schema</a>.
```

Comment:
This should rather be specified where you write about POST and PUT bodies — _we_ ensure they match the schema here :)

Reply:
correct! moving